### DISCUSSION TITLE
Fips certification on openssl version needs clarity.

### DISCUSSION THREAD

**Discussion by chaitu-venky** [openssl/openssl#23084] [Q&A] [answered]
As per the openssl document openssl only 3.0.0 & 3.0.8 versions are fips validated: https://www.openssl.org/source/
I see that Openssl 3.1 openssl version is submitted for 140-3 certification, Can i assume that openssl 3.1 version as FIPS 140-2 validated/certified ? 
My understanding is that once Openssl 3.X got certified, the above versions(openssl 3.X+) also holds the same certificate, until explicitly the documents says so. 
I wanted to consume the latest version openssl 3.2 release with FIPS. It will be really helpful if my above questions get answered. so that based on that i can take the FIPS validated/certified openssl.

**Comment by mattcaswell** [comment, marked as answer]
> As per the openssl document openssl only 3.0.0 & 3.0.8 versions are fips validated: https://www.openssl.org/source/

Only the explicit versions indicated are validated (3.0.0 and 3.0.8). 3.1 is not yet validated, although it is the intention that it will achieve 140-3 validation.

> I see that Openssl 3.1 openssl version is submitted for 140-3 certification, Can i assume that openssl 3.1 version as FIPS 140-2 validated/certified ?

3.1 will not be submitted  for 140-2 validation. Only releases in the 3.0 line will be validated for 140-2 (i.e. you cannot assume that 3.1 has 140-2 validation).

> My understanding is that once Openssl 3.X got certified, the above versions(openssl 3.X+) also holds the same certificate, until explicitly the documents says so.

This is incorrect. Only the explicit versions indicated are validated.

> I wanted to consume the latest version openssl 3.2 release with FIPS.

3.2 is not validated at this time.

It is perfectly possible (and in fact a recommended approach) to use the latest version of OpenSSL (e.g. 3.2.0) with a validated version of the FIPS provider (e.g. 3.0.8). Refer to the README-FIPS.md file in the 3.2.0 download. Also see my talk on the FIPS provider here:

https://www.youtube.com/watch?v=geAtEXbHaFg

**Comment by joelguittet** [comment]
Hello @mattcaswell 

I found this discussion looking for information about FIPS 140-3 certification.
Any news about this ? I see it is scheduled for 2024 here: https://www.openssl.org/blog/blog/2023/10/12/osslfips-timeline/index.html

Thanks for the update!
Joel

**Reply by paulidale** [reply]
The FIPS lab we were using is no longer accredited.  That's going to stall the 140-3 validation I suspect.

**Reply by joelguittet** [reply]
Ok thanks @paulidale for this update
